packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -166,13 +197,74 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
             IUniswapV3Pool(wPowerPerpPool).fee(),
             _wPowerPerpAmount,
             amountToFlashswap,
-            uint8(FLASH_SOURCE.FLASH_W_MINT),
+            uint8(CALLBACK_SOURCE.FLASH_W_MINT),
             abi.encodePacked(_vaultId, amountToFlashswap, _collateralAmount, _wPowerPerpAmount)
         );
 
         emit FlashswapWMint(msg.sender, _vaultId, _wPowerPerpAmount, amountToFlashswap, _collateralAmount);
     }
 
+    /**
+     * @notice flash close position and buy long squeeth
+     * @dev this function
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmountToBurn amount of WPowerPerp to burn
+     * @param _wPowerPerpAmountToBuy amount of WPowerPerp to buy
+     * @param _collateralToWithdraw amount of collateral to withdraw from vault
+     * @param _maxToPay max amount of collateral to pay for WPowerPerp token
```

Comment:
this is max to receive for sum of _wPowerPerpAmountToBuy and _wPowerPerpAmountToBurn, might be good to be explicit